lib/utils/remote-build.ts

```diff
@@ -300,38 +353,56 @@ function createRemoteBuildRequest(
 async function getRemoteBuildStream(
 	build: RemoteBuild,
 ): Promise<NodeJS.ReadWriteStream> {
-	const tarStream = await getTarStream(build);
-	const visuals = await import('resin-cli-visuals');
-	const uploadSpinner = new visuals.Spinner(
-		'Uploading source package to balena cloud',
+	const builderUrl = await getBuilderEndpoint(
+		build.baseUrl,
+		build.owner,
+		build.app,
+		build.opts,
 	);
-	const exitOnError = (error: Error): never => {
-		uploadSpinner.stop();
+
+	let uploadSpinner = {
+		stop: () => {
+			/* noop */
+		},
+	};
+	let exitOnError = (error: Error) => {
 		return exitWithExpectedError(error);
 	};
+	// We only show the spinner when outputting to a tty
+	if (process.stdout.isTTY) {
```

Comment:
Note that sniffing for TTYs is not always sufficient for determining if you are in a CI environment -- Jenkins builds, as a very pertinent example, allow users to turn tty emulation on/off in the Jenkinsfile.  It would be nice to be able to explicitly disable spinners and other visuals with a flag or environment variable.

Reply:
Thanks for the heads up, I agree that this would certainly be better, but it's a much larger PR, as we'll need to handle all commands which can be started with `balena push`. I'd rather tackle this seperately. Given that it's possible to force non-tty execution even within a tty, this can be worked around until that PR is merged. I've created an issue here to track it: https://github.com/balena-io/balena-cli/issues/1378